ProcessLib/BoundaryCondition/HCOpenBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,117 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "GenericNaturalBoundaryConditionLocalAssembler.h"
+#include "MeshLib/Elements/MapBulkElementPoint.h"
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "NumLib/Function/Interpolation.h"
+#include "ProcessLib/Parameter/MeshNodeParameter.h"
+#include "ProcessLib/Process.h"
+
+namespace ProcessLib
+{
+struct HCOpenBoundaryConditionData
+{
+    Parameter<double> const& boundary_permeability;
+    MeshLib::PropertyVector<std::size_t> const bulk_face_ids;
+    MeshLib::PropertyVector<std::size_t> const bulk_element_ids;
+    Process const& process;
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+class HCOpenBoundaryConditionLocalAssembler final
+    : public GenericNaturalBoundaryConditionLocalAssembler<
+          ShapeFunction, IntegrationMethod, GlobalDim>
+{
+    using Base = GenericNaturalBoundaryConditionLocalAssembler<
+        ShapeFunction, IntegrationMethod, GlobalDim>;
+    using NodalVectorType = typename Base::NodalVectorType;
+    using NodalMatrixType = typename Base::NodalMatrixType;
+
+public:
+    /// The neumann_bc_term factor is directly integrated into the local
+    /// element matrix.
+    HCOpenBoundaryConditionLocalAssembler(
+        MeshLib::Element const& e,
+        std::size_t const local_matrix_size,
+        bool const is_axially_symmetric,
+        unsigned const integration_order,
+        HCOpenBoundaryConditionData const& data)
+        : Base(e, is_axially_symmetric, integration_order),
+          _data(data),
+          _local_matrix_size(local_matrix_size)
+    {
+    }
+
+    void assemble(std::size_t const mesh_item_id,
+                  NumLib::LocalToGlobalIndexMap const& dof_table_boundary,
+                  double const t, const GlobalVector& x, GlobalMatrix& /*K*/,
+                  GlobalVector& b, GlobalMatrix* /*Jac*/) override
+    {
+        NodalVectorType _local_rhs(_local_matrix_size);
+        _local_rhs.setZero();
+        // Get element nodes for the interpolation from nodes to
+        // integration point.
+        NodalVectorType const boundary_permeability_node_values =
+            _data.boundary_permeability.getNodalValuesOnElement(Base::_element,
+                                                                t);
+        auto surface_element_normal =
+            MeshLib::FaceRule::getSurfaceNormal(&(Base::_element));
```

Comment:
Move to ctor; store Eigen::Vector3d.

Reply:
I am not sure about the "prettiness" of my updated solution... (?)